src/modules/mim/firewall.json

```diff
@@ -1,41 +1,187 @@
-{ 
-  "name": "FirewallModel", 
+{
+  "name": "FirewallModel",
   "type": "MimModel",
-  "contents": [ 
-    { 
-      "name": "Firewall", 
-      "type": "mimComponent", 
+  "contents": [
+    {
+      "name": "Firewall",
+      "type": "mimComponent",
       "contents": [
-        { 
-          "name": "firewallState", 
-          "type": "mimObject", 
+        {
+          "name": "firewallState",
+          "type": "mimObject",
           "desired": false,
           "schema": {
-             "type": "enum",
-             "valueSchema": "integer",
-             "enumValues": [
-               { 
-                 "name": "unknown",
-                 "enumValue": 0
-               },
-               {
-                 "name": "enabled",
-                 "enumValue": 1
-               },
-               {
-                 "name": "disabled",
-                 "enumValue": 2
-               }
-             ]
-           }
+            "type": "enum",
+            "valueSchema": "integer",
+            "enumValues": [
+              {
+                "name": "unknown",
+                "enumValue": 0
+              },
+              {
+                "name": "enabled",
+                "enumValue": 1
+              },
+              {
+                "name": "disabled",
+                "enumValue": 2
+              }
+            ]
+          }
         },
-        { 
-          "name": "firewallFingerprint", 
-          "type": "mimObject", 
+        {
+          "name": "firewallFingerprint",
+          "type": "mimObject",
           "desired": false,
           "schema": "string"
+        },
+        {
+          "name": "createRule",
```

Comment:
>createRule [](http://example.com/codeflow?start=19&length=10)

Not right for an object name. This name is for an action, an object name should be for an object.

Reply:
desiredFirewallRule?

Reply:
Add "desired" to name to tell that this is not a reported object.